src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -124,13 +130,185 @@ Status RedisStoreClient::AsyncDelete(const std::string &table_name,
   return shard_context->RunArgvAsync(args, delete_callback);
 }
 
+Status RedisStoreClient::AsyncBatchDelete(const std::string &table_name,
+                                          const std::vector<std::string> &keys,
+                                          const StatusCallback &callback) {
+  auto finished_count = std::make_shared<int>(0);
+  int size = keys.size();
+  for (auto &key : keys) {
+    auto done = [finished_count, size, callback](const Status &status) {
+      ++(*finished_count);
+      if (*finished_count == size) {
+        callback(Status::OK());
+      }
+    };
+    RAY_CHECK_OK(AsyncDelete(table_name, key, done));
+  }
+  return Status::OK();
+}
+
 Status RedisStoreClient::AsyncDeleteByIndex(const std::string &table_name,
                                             const std::string &index_key,
                                             const StatusCallback &callback) {
-  RAY_CHECK(0) << "Not implemented! Will implement this function in next PR.";
+  std::string match_pattern = index_key + table_name + "*";
```

Comment:
1) minor: putting table name in the first would be more human readable and easier for search.
2) also add separator.
3) it seems that you didn't change the key of `PutWithIndex`

Reply:
fixed